fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -464,6 +467,11 @@ public void dropDb(String dbName, boolean isForceDrop) throws DdlException, Meta
                 throw new MetaNotFoundException("Database not found");
             }
             db = this.fullNameToDb.get(dbName);
+            if (!isForceDrop && db.getTemporaryTables().size() > 0) {
+                throw new DdlException("The database [" + dbName + "] " +
+                        "cannot be dropped because there are still some temporary tables in it. " +
+                        "If you want to forcibly drop, please use \"DROP DATABASE <database> FORCE.\"");
```

Comment:
Should also consider tmp table cannot go into recycle bin?

Reply:
> Should also consider tmp table cannot go into recycle bin?

already considered, plz see `Database::unprotectDropTemporaryTable`, no matter what ,we won't recyclyTable